File: backend/src/main/java/com/equisoft/dca/backend/user/model/Feature.java
package com.equisoft.dca.backend.user.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum Feature implements BaseEnum<Integer> {

	SYSTEM_SETTINGS(1, "System settings"),
	ORGANIZATIONS_AND_USERS(2, "Organizations and users"),
	PROJECT_MANAGEMENT(3, "Project management"),
	PROJECT_DATA_MAPS(4, "Project datamaps"),
	PROJECT_TABLES(5, "Project tables"),
	SYSTEM_MANAGEMENT(6, "System management"),
	SYSTEM_CODE_ANALYSIS(7, "System code analysis"),
	SYSTEM_DATA_OBJECTS(8, "System data objects"),
	SYSTEM_TRANSACTIONS(9, "System transactions");

	private final int id;

	private final String name;

	Feature(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: react/src/lib/domain/entities/Feature.ts
import BaseEnum from './BaseEnum';

export default class Feature extends BaseEnum {
    public static SYSTEM_SETTINGS = new Feature('1', 'user:featuresEnum:systemSettings');
    public static ORGANIZATIONS_AND_USERS = new Feature('2', 'user:featuresEnum:organizationsUsers');
    public static PROJECT_MANAGEMENT = new Feature('3', 'user:featuresEnum:projectManagement');
    public static PROJECT_DATA_MAPS = new Feature('4', 'user:featuresEnum:projectDatamaps');
    public static PROJECT_TABLES = new Feature('5', 'user:featuresEnum:projectTables');
    public static SYSTEM_MANAGEMENT = new Feature('6', 'user:featuresEnum:systemManagement');
    public static SYSTEM_CODE_ANALYSIS = new Feature('7', 'user:featuresEnum:systemCodeAnalysis');
    public static SYSTEM_DATA_OBJECTS = new Feature('8', 'user:featuresEnum:systemDataObjects');
    public static SYSTEM_TRANSACTIONS = new Feature('9', 'user:featuresEnum:systemTransaction');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}
